python/ray/serve/_private/deployment_state.py

```diff
@@ -1942,6 +1944,22 @@ def check_curr_status(self) -> Tuple[bool, bool]:
                 )
                 return False, any_replicas_recovering
 
+        if failed_to_start_count > self._prev_replica_constructor_retry_counter:
```

Comment:
Why is `_prev_replica_constructor_retry_counter` necessary? Can we just always call `update_message` and change it to a no-op if the message is the same?

Reply:
yeah you're right